### PR TITLE
Add APL-related items

### PR DESCRIPTION
**Pull request by abrudz**
## What does this PR do?
Adds APL-related items and fixes various smaller things

### Description
Adds APL-related items and fixes various smaller things
### Why is this valuable (or not)?
Very few APL-related items were present. This adds many more.
### How do we know it's really free?
Most of these have listed licenses or are freely available websites.
### For book lists, is it a book? For course lists, is it a course? etc.
Each type of item is in its appropriate place.
## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md)
- [x] Search for duplicates.
- [x] Include author(s) and platform where appropriate.
- [x] Put lists in alphabetical order, correct spacing.
- [x] Add needed indications (PDF, access notes, under construction)

## Follow-up

- [x] Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

File: books/free-programming-books-langs.md [hunk 1]
```diff
@@ -268,11 +268,12 @@ That section got so big, we decided to split it into its own file, the [BY SUBJE
 
 ### APL
 
-* [A Practical Introduction to APL1 & APL2](http://robertson.uk.net) - Graeme Donald Robertson (PDF)
-* [A Practical Introduction to APL3 & APL4](http://robertson.uk.net) - Graeme Donald Robertson (PDF)
+* [A Practical Introduction to APL1 & APL2](http://robertson.uk.net) - Graeme Donald Robertson ([PDF](http://robertson.uk.net/Files/APL1&2.pdf))
```
Comment (eshellman) [inline books/free-programming-books-langs.md:271]: one link per entry please - probably the pdf is best.

File: courses/free-courses-fr.md [hunk 2]
```diff
@@ -25,6 +26,13 @@
 * [Cours d'Algorithmie](https://www.youtube.com/playlist?list=PLrSOXFDHBtfGy7xYmf5LlNr8f-niDkf7_) - Formation Video
 
 
+### APL
+
+* [Découvrez le langage APL : 1er partie avec fonctions de base et exercices corrigés](https://www.youtube.com/watch?v=Nxq1BUUXobM) - Eric Schrafstetter
```
Comment (eshellman) [inline courses/free-courses-fr.md:31]: For youtube, we prefer "playlist" links and the name of the producer/creator. https://github.com/EbookFoundation/free-programming-books/blob/master/CONTRIBUTING.md#platforms-and-access-notes  Is the a playlist that includes all three?

File: more/free-programming-cheatsheets.md [hunk 3]
```diff
@@ -36,6 +37,21 @@
 * [Tensorflow](#tensorflow)
 
 
+### APL
+
+* [A reference card for GNU APL](https://github.com/jpellegrini/gnu-apl-refcard/blob/master/aplcard.pdf) - jpellegrini (PDF)
+* [CheatSheet - Binding Strengths](https://docs.dyalog.com/latest/CheatSheet%20-%20Binding%20Strengths.pdf) - Dyalog (PDF)
```
Comment (eshellman) [inline more/free-programming-cheatsheets.md:43]: maybe there's one link for the dyalog cheatsheets?

File: courses/free-courses-en.md [hunk 4]
```diff
@@ -156,6 +157,15 @@
 * [Programming Mobile Services for Android Handheld Systems: Concurrency](https://www.coursera.org/course/posaconcurrency)
 
 
+### APL
+
+* [APL Course](https://course.dyalog.com) - Dyalog (:construction: *in process*)
+* [APL Cultivation](https://apl.wiki/APL_Cultivation) - Adám Brudzewsky
```
Comment (eshellman) [inline courses/free-courses-en.md:163]: use https://aplwiki.com/wiki/APL_Cultivation The redirect id funky.

File: courses/free-courses-en.md [hunk 5]
```diff
@@ -156,6 +157,15 @@
...
+* [Dyalog APL Tutor](https://tutorial.dyalog.com) - Dyalog
+* [Learn APL with Neural Networks](https://www.youtube.com/playlist?list=PLgTqamKi1MS3p-O0QAgjv5vt4NY5OgpiM) - Rodrigo Girão Serrão 
+* [Reinforcement Learning From The Ground Up](https://romilly.github.io/o-x-o) - Romilly Cocking (:construction: *in process*)
```
Comment (eshellman) [inline courses/free-courses-en.md:166]: this is a book.

**Review by eshellman — CHANGES_REQUESTED** [review]
This is a fantastic contribution! See if you can address the comments.

File: more/free-programming-cheatsheets.md (hunk 3 above)
Comment (abrudz) [reply more/free-programming-cheatsheets.md:43]: There's only https://docs.dyalog.com/latest which probably isn't something you want to link to.

Comment (abrudz) [reply more/free-programming-cheatsheets.md:43]: @eshellman Is this acceptable as-is?

Comment (eshellman) [reply more/free-programming-cheatsheets.md:43]: how about https://www.dyalog.com/documentation_180.htm#CHEAT ?

Comment (abrudz) [reply more/free-programming-cheatsheets.md:43]: That's version-specific and will be out of date within a month. Is https://docs.dyalog.com#CHEAT acceptable even though it is a redirect?

Comment (eshellman) [reply more/free-programming-cheatsheets.md:43]: yes, that looks good

**Comment by eshellman** [comment]
Thanks, and welcome to free-programming-books!

**Pull request merged**